tembo-operator/tests/integration_tests.rs

```diff
@@ -65,6 +65,112 @@ mod test {
     const TIMEOUT_SECONDS_POD_DELETED: u64 = 300;
     const TIMEOUT_SECONDS_COREDB_DELETED: u64 = 300;
 
+    /// Struct to contain many commonly used test resources
+    /// 
+    /// Most if not all tests use all of the fields here, or refer to them
+    /// in some manner. This struct helps combine everything together.
+    struct TestClass {
+        name: String,
+        namespace: String,
+        client: Client,
+        state: State,
+        context: Arc<Context>,
+        pods: Api<Pod>,
```

Comment:
I really like this concept and it's maybe something we can go back and adjust for all tests.  

The only issue is that these three objects in the struct are not being used.  I suggest to comment them out for now until they are used.

Reply:
Yeah... I was kinda hoping there was a way to just suppress those specific warnings because there's full intent to use those.

Reply:
Now that I reexamine, I think a better solution than commenting these out would be to provide a get interface rather than reaching deep into the struct. These elements in particular are often used directly in the other tests, so need a better assignment paradigm. That would solve both problems at once. I'll patch that in.

Reply:
On second thought, that just creates more functions that will remain "unused" and add back into the code duping issue. I'll just convert the basic-cnpg test which uses these.